src/main/java/com/_up/megastore/services/implementations/ProductService.java

```diff
@@ -57,20 +57,36 @@ private String saveProductImage(MultipartFile multipartFile) {
     private Product getVariantOf(UUID variantOfId) {
         return variantOfId != null ? findProductByIdOrThrowException(variantOfId) : null;
     }
+    @Override
+
+
+
     public ProductResponse updateProduct(UUID productId, UpdateProductRequest updateProductRequest, MultipartFile multipartFile){
         Product product = this.findProductByIdOrThrowException(productId);
         Category category = categoryService.findCategoryByIdOrThrowException(updateProductRequest.categoryId());
-        Product variantOf = this.getVariantOf(updateProductRequest.variantOfId());
-        String imageURL = saveProductImage(multipartFile);
+
+        ifVariantOfExistsUpdateProductVariantOf(updateProductRequest, product);
+
+        ifImageURLExistsUpdateProductImageURL(multipartFile, product);
 
         product.setName(updateProductRequest.name());
         product.setDescription(updateProductRequest.description());
         product.setPrice(updateProductRequest.price());
-        product.setImageURL(imageURL);
-        product.setCategory(category);
-        product.setVariantOf(variantOf);
 
+        product.setCategory(category);
 
-        return ProductMapper.toProductResponse(productRepository.save(product) );
+        return ProductMapper.toProductResponse( productRepository.save(product) );
+    }
+    public void ifVariantOfExistsUpdateProductVariantOf(UpdateProductRequest updateProductRequest, Product product){
+        if(updateProductRequest.variantOfId() != null){
+            Product variantOf = this.getVariantOf(updateProductRequest.variantOfId());
+            product.setVariantOf(variantOf);
+        }
+    }
+    public void ifImageURLExistsUpdateProductImageURL(MultipartFile multipartFile, Product product){
+        if(saveProductImage(multipartFile) != null){
+            String imageURL = saveProductImage(multipartFile);
+            product.setImageURL(imageURL);
+        }
     }
```

Comment:
La lógica está bien pero, de nuevo, los espacios. Deberíamos respetar el formato del código en todas las clases

```suggestion
    }
    public void ifVariantOfExistsUpdateProductVariantOf(UpdateProductRequest updateProductRequest, Product product) {
        if(updateProductRequest.variantOfId() != null) {
            Product variantOf = this.getVariantOf(updateProductRequest.variantOfId());
            product.setVariantOf(variantOf);
        }
    }
    
    public void ifImageURLExistsUpdateProductImageURL(MultipartFile multipartFile, Product product) {
        if (saveProductImage(multipartFile) != null) {
            String imageURL = saveProductImage(multipartFile);
            product.setImageURL(imageURL);
        }
    }
```
Los métodos tienen que separarse entre sí por una sola línea, sino nos queda todo junto